### PR TITLE
Add dev.bragefuglseth.Keypunch

### PR DESCRIPTION
**Pull request by bragefuglseth**
Keypunch is an app for practicing typing on a keyboard.

### Please confirm your submission meets all the criteria

- [x] Please describe your application briefly.
- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I have [built][build] and tested the submission locally.
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] The domain used for the application ID is controlled by the application developers either directly or through the code hosting (e.g. GitHub, GitLab, SourceForge, etc.). The [application id guidelines][app-id] are followed.
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission
[app-id]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id

### CODE REVIEW

**Comment by bragefuglseth** [comment]
bot, build dev.bragefuglseth.Keypunch

**Comment by bragefuglseth** [comment]
bot, build

**Comment by bragefuglseth** [comment]
bot, build dev.bragefuglseth.Keypunch

**Comment by bragefuglseth** [comment]
bot, build dev.bragefuglseth.Keypunch

**Comment by bragefuglseth** [comment]
Ah. The ID still has .Devel from the development version, since I forgot to change that.

**Comment by bragefuglseth** [comment]
bot, build dev.bragefuglseth.Keypunch

**Comment by bragefuglseth** [comment]
note to self: don't forget that you should test the production manifest locally too next time you submit something.

**Comment by bragefuglseth** [comment]
works great here

File: dev.bragefuglseth.Keypunch.json [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+	  "cleanup" : ["*"],
+	  "sources" : [
+	    {
+	      "type" : "git",
+	      "url" : "https://gitlab.gnome.org/jwestman/blueprint-compiler",
+	      "tag" : "v0.12.0"
+	    }
+	  ]
+	},
+        {
```
Comment (bbhtt) [inline dev.bragefuglseth.Keypunch.json:43]: Indentation is broken for this module.

Comment (bragefuglseth) [reply dev.bragefuglseth.Keypunch.json:43]: Thanks for the heads-up.

File: dev.bragefuglseth.Keypunch.json [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        "cargo-sources.json"
+      ]
+    }
+  ],
+  "runtime": "org.gnome.Platform",
+  "runtime-version": "46",
+  "sdk": "org.gnome.Sdk",
+  "sdk-extensions": [
+    "org.freedesktop.Sdk.Extension.rust-stable"
+  ]
```
Comment (bbhtt) [inline dev.bragefuglseth.Keypunch.json:64]: Looks like they to the bottom?

File: dev.bragefuglseth.Keypunch.json [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    "*.a"
+  ],
+  "command": "keypunch",
+  "finish-args": [
+    "--share=ipc",
+    "--socket=fallback-x11",
+    "--device=dri",
+    "--socket=wayland"
+  ],
+  "id": "dev.bragefuglseth.Keypunch",
```
Comment (bbhtt) [inline dev.bragefuglseth.Keypunch.json:23]: This should be on top

File: dev.bragefuglseth.Keypunch.json [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    "*.la",
+    "*.a"
+  ],
+  "command": "keypunch",
+  "finish-args": [
+    "--share=ipc",
+    "--socket=fallback-x11",
+    "--device=dri",
+    "--socket=wayland"
+  ],
```
Comment (bbhtt) [inline dev.bragefuglseth.Keypunch.json:22]: Before toplevel cleanup.

File: dev.bragefuglseth.Keypunch.json (hunk 2 above)
Comment (bragefuglseth) [reply dev.bragefuglseth.Keypunch.json]: The formatter I used apparently alphabetized all the fields. I've fixed it now.

**Comment by bbhtt** [comment]
bot, build dev.bragefuglseth.Keypunch

**Pull request closed without merging**